docs/zkapps/tutorials/interacting-with-zkapps-server-side.mdx

```diff
@@ -10,279 +16,268 @@ zkApp programmability is not yet available on the Mina Mainnet, but zkApps can n
+// Programmatic deploy:
+//   Besides the CLI, you can also create accounts programmatically. This is useful if you need
+//   more custom account creation - say deploying a zkApp to a different key than the fee payer
+//   key, programmatically parameterizing a zkApp before initializing it, or creating Smart
+//   Contracts programmatically for users as part of an application.
+await deploy(deployerPrivateKey, zkAppPrivateKey, zkapp, verificationKey);
 
-<!-- note: I'm commenting this out since setting `provedState` with `init()` doesn't work yet anyway because of the missing verification key protocol bug.
+await loopUntilAccountExists({
+  account: zkAppPublicKey,
+  eachTimeNotExist: () =>
+    console.log('waiting for zkApp account to be deployed...'),
+  isZkAppAccount: true,
+});
 
-:::info
+let num = (await zkapp.num.fetch())!;
+console.log(`current value of num is ${num}`);
+
+// ----------------------------------------------------
+...
+```
 
-There is a `provedState` property on accounts present in the protocol, but yet to be exposed in o1js, that can be used to assert in a proof that the smart contract has been initialized.
+To do this, reuse the helper function `loopUntilAccountExists()` from `utils.js`. This time, pass in `isZkappAccount: true` checks if the account exists and that there is a verification key on the account. An existing verification key indicates that the zkApp has been successfully deployed.
 
-This `provedState` property starts out `false` when a contract is deployed. It is set to true whenever the _entire state_ of the smart contract is overwritten by a proof. TODO. More details will be released as this feature is made available.
+The smart contract was already deployed with `zk deploy` so a programmatic deploy is not required and is commented out here. If you want to see how this works, or it's useful for your application, see the code in [utils.ts](https://github.com/o1-labs/docs2/blob/main/examples/zkapps/interacting-with-zkApps-server-side/src/utils.ts).
 
-::: -->
+After the zkApp has been deployed, fetch the current value of `zkapp.num` (the on-chain defined on the `SmartContract`) and log it. If this is the first time you have run this script, the value is `3` because that's how it is set in the smart contract's `init()` function. The `init()` function is called automatically during the first deploy (not during re-deploys).
 
-### Calling update on our transaction
+### Call an update on the transaction
 
-And lastly, let's now send an update to our transaction. If the zkApp was just initialized, this will call update on the newly initialized account. Otherwise, it will call update on whatever the current account state happens to be.
+Finally, send an update to the transaction. If the zkApp was just initialized, this calls an update on the newly initialized account. Otherwise, it calls an update on whatever the current account state happens to be.
```

Comment:
I find this block to be a little hard to understand. I might call this section something like "Sending an Update Transaction." And then continue that phrasing thought the rest of the paragraph.